app/src/main/java/space/taran/arknavigator/mvp/model/dao/Resource.kt

```diff
@@ -1,16 +1,20 @@
 package space.taran.arknavigator.mvp.model.dao
 
+import androidx.room.ColumnInfo
 import androidx.room.Entity
+import androidx.room.Index
 import androidx.room.PrimaryKey
 import space.taran.arknavigator.mvp.model.repo.index.ResourceId
 import space.taran.arknavigator.mvp.model.repo.index.ResourceMeta
 import space.taran.arknavigator.utils.Milliseconds
 import space.taran.arknavigator.utils.StringPath
 import java.nio.file.Path
 
-@Entity
+@Entity(indices = [Index(value = ["id"], unique = true)])
 data class Resource(
-    @PrimaryKey(autoGenerate = false)
+    @PrimaryKey(autoGenerate = true)
+    val index: Int = 0,
```

Comment:
it is used to order the file list in resource grid view. before that we have order based on `id` which is `resorceId`. After updating image, the `id` of the image is changed due to change in file size. attached the reasons.
![Screenshot from 2022-07-08 11-44-13](https://user-images.githubusercontent.com/74892023/177928837-ba54e448-d878-42b4-b9b8-6c45fbbed6da.png)
Now, If updated `id` has greater value than other `Ids` then it moves to bottom in DB. If the `Id` is lesser than other `Ids` then it moves to top. 
Due to this, image list in UI will be change without any hint and it will be tricky to understand by user.
I have faced this issue. I have 3 images and updated image went to last in the list when app is killed and reopened.
So I added this `index` column to removed dependency from `id` and an order is also maintain after update the image.

Reply:
OK then, I will removed the index column and let you know.

Reply:
@mdrlzy actually, we might want keeping order as @pgpathan22 implemented initially. But at this point, we will not implement it, because first of all we need to move index into persisted/replicated storage (#142) and that task would change DB scheme significantly, so better to not interfere with this for now.